### PR TITLE
solved day9 problems

### PR DESCRIPTION
**Pull request by peter-sushko**
Hi Tim,

Here is day 9. Thanks for looking into this!

### CODE REVIEW

File: day9/part_1.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    
+    Example: 
+    Given the following list, predict the next value:
+    10 13 16 21 30 45
+    
+    We solve this by looking at the lists containing differences.
+    Notice that the solution to "?" is 45 + 23 = 68.
+    The we got 23 by adding up 15 and 8. Continuing with this logic,
+    we get that ? equals to 45 + 15 + 6 + 2 + 0
+    which are the last elements of each array of differences.
```
Comment (tshur) [inline day9/part_1.py:20]: nice observation!

File: day9/part_1.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    
+    10  13  16  21  30  45  ?
+       3   3   5   9  15  23
+         0   2   4   6   8
+           2   2   2   2
+             0   0   0
+    """
+    prediction = 0
+    list_char = row[:-1].split(' ')
```
Comment (tshur) [inline day9/part_1.py:29]: i liked when you had a separate function to do the parsing. it helps for extensibility and also decoupling

File: day9/part_1.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    current_list = [int(number) for number in list_char]
+    while not all(x == 0 for x in current_list):
+        prediction += current_list[-1]
+        current_list = find_differences(current_list)
+    return prediction
+
+def find_differences(input_list: list):
```
Comment (tshur) [inline day9/part_1.py:36]: consider typing the return value too...

i think def fn(arg: argtype) -> rettype: or something?

File: day9/part_1.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    """Given a list, output a list containing differences."""
+    output_list = []
+    for value in range(len(input_list)-1):
+        output_list.append(input_list[value + 1] - input_list[value ])
```
Comment (tshur) [inline day9/part_1.py:40]: nit: extra space near the end of this line -- do you have an autoformatter?

File: day9/part_2.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    current_list = [int(number) for number in list_char]
+    prediction = current_list[0]
+    count = 1
+    while not all(x == 0 for x in current_list):
+        current_list = find_differences(current_list)
+        prediction += current_list[0] * (-1)**count
+        count += 1
+    return prediction
+
+def find_differences(input_list: list):
```
Comment (tshur) [inline day9/part_2.py:22]: maybe you're intentionally not connecting the part1 and part2 files but some functions like this are exactly the same between the two

File: day9/part_2.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    consecutive elements. Starting with the first element, it alternately adds and subtracts the 
+    differences to compute the prediction.   
+    """
+    list_char = row[:-1].split(' ')
+    current_list = [int(number) for number in list_char]
+    prediction = current_list[0]
+    count = 1
+    while not all(x == 0 for x in current_list):
+        current_list = find_differences(current_list)
+        prediction += current_list[0] * (-1)**count
```
Comment (tshur) [inline day9/part_2.py:18]: i think i actually found the example to have an error in it -- i think i just summed the [0] index or something for my correct answer but i didn't flip flop

**Review by tshur — APPROVED** [review]

File: day9/part_1.py (hunk 4 above)
Comment (peter-sushko) [reply day9/part_1.py:40]: I dont. Which one do you recommend?

File: day9/part_2.py (hunk 5 above)
Comment (peter-sushko) [reply day9/part_2.py:22]: yes, I keep them as 2 separate python modules on purpose. Each module solves 1 problem

**Pull request merged**